addons/website/tools.py

```diff
@@ -22,37 +25,96 @@ def get_video_embed_code(video_url):
     # Regex for few of the widely used video hosting services
     ytRegex = r'^(?:(?:https?:)?\/\/)?(?:www\.)?(?:youtu\.be\/|youtube(-nocookie)?\.com\/(?:embed\/|v\/|watch\?v=|watch\?.+&v=))((?:\w|-){11})(?:\S+)?$'
     vimeoRegex = r'\/\/(player.)?vimeo.com\/([a-z]*\/)*([0-9]{6,11})[?]?.*'
-    dmRegex = r'.+dailymotion.com\/(video|hub|embed)\/([^_]+)[^#]*(#video=([^_&]+))?'
+    dmRegex = r'.+dailymotion.com\/(video|hub|embed)\/([^_?]+)[^#]*(#video=([^_&]+))?'
     igRegex = r'(.*)instagram.com\/p\/(.[a-zA-Z0-9]*)'
```

Comment:
This regex does not match `https://www.instagram.com/p/B_8Pt4LAxzg/`

Reply:
Note: for all these regexes, @JKE-be and I think we should probably try to share them with the JS instead of being duplicated. I already merged at least one fix where I changed the regex in JS and not in python or the other way around: https://github.com/odoo/odoo/commit/223038682f59f48e221a6e74981b3723fd31f3e8. Also the variables are actually camelCased in python because we copied them from the JS :/. Maybe we can try to unify all of that, even if it means having to do rpc from the JS to ask the server to process the video URLs? Could you think about that @bbh-odoo ? Thanks!